### ISSUE TITLE
Should the post-create terminal close automatically if it exited cleanly

### ISSUE THREAD

**Issue by connor4312** [microsoft/vscode-remote-release#4537]
Testing #4502

Right now the terminal is always shown where the command is run, and the user is prompted to press any key to close the terminal after the command finishes. This mandates manual user interaction for thing they _usually_ would not care about.

I think it would be a better experience to automatically close the terminal when the run completes if the command was not interrupted and exited with a 0.

**Comment by chrmarti** [comment]
That might make the UI a little noisy. I'm imagining a short command would show the terminal for an equally short duration.

Whether or not the user is interested in the output might depend on the command being run. E.g., a `npm install` might be more interesting because that tells me I should not expect everything to fully work before the packages are installed.

**Comment by bamurtaugh** [comment]
I tend to agree with what @chrmarti mentions - an automatic close may be expected for some but unexpected/jarring for others (i.e. if I wasn't paying close attention, I might not know what happened, or might think something closed by accident). 

I also agree in the value of seeing the output of certain commands - in Codespaces, now being able to tell when `npm install` is complete (since I can see the Oryx spinner in the status bar and navigate to the logs/output) is really helpful.

**Comment by thaney071** [comment]
If the window won't close automatically then it should be able to handle reviving input. Post commands shouldn't be required to do a read or be long running as mentioned in this #4539. 

Additionally just my opinion here, but most of the other developers working with the devcontainer don't care and shouldn't care about the post commands running. It is helpful if something goes wrong in setup, but most of them just start the container and don't know as much about how it is working. We have a subset of developers that maintain the dev environment so not every one has to. I would prefer if the window just closed on success. That way if is does fail they can send the log output, otherwise nobody will care to see the output.

**Comment by tapionx** [comment]
+1

**Comment by ericriff** [comment]
I also would like an option to make this go away by itself if the cmds executed correctly. For the developer working on the devcontainer itself, it might be useful. But for the end user, it is just noise. Closing this is just an extra step they must perform to enable their environment.

**Comment by mjnowen** [comment]
+1 to silence this terminal by default.

**Comment by MeoMix** [comment]
Is there a way to express through configuration that the terminal should close - even if it's not the default behavior? I can see how it would be contentious as default behavior, but I would love an opt-in setting.

**Comment by gboeer** [comment]
I also think, having some option to continue after the post command would be great.

**Comment by mjnowen** [comment]
@chrmarti - could this request be reviewed again? thank you.

**Comment by Nummer42O** [comment]
> I can see how it would be contentious as default behavior, but I would love an opt-in setting.

Yes I would also really see opt-in as the way to go here. I think structuring it like the presentation property of tasks would be helpful as people already know how to use that.

**Comment by kvokka** [comment]
+1

IMHO, if it succeeds, it should be an option to make it silent for all lifecycle scripts.

**Comment by mjnowen** [comment]
+1

Please implement this feature. It would be most useful.
